src/GraphNavigator/DeserializationGraphNavigator.php

```diff
@@ -209,15 +209,21 @@ public function accept($data, ?array $type = null)
                         continue;
                     }
 
-                    $this->context->pushPropertyMetadata($propertyMetadata);
+                    /** Metadata changes based on context, should not be cached  */
+                    $contextSpecificMetadata = $propertyMetadata;
+                    if (null !== $this->context->getPropertyNamingStrategy()) {
+                        $contextSpecificMetadata = clone $propertyMetadata;
+                        $contextSpecificMetadata->serializedName = $this->context->getPropertyNamingStrategy()->translateName($propertyMetadata);
+                    }
+
+                    $this->context->pushPropertyMetadata($contextSpecificMetadata);
                     try {
-                        $v = $this->visitor->visitProperty($propertyMetadata, $data);
-                        $this->accessor->setValue($object, $v, $propertyMetadata, $this->context);
+                        $v = $this->visitor->visitProperty($contextSpecificMetadata, $data);
+                        $this->accessor->setValue($object, $v, $contextSpecificMetadata, $this->context);
                     } catch (NotAcceptableException $e) {
-                        if (true === $propertyMetadata->hasDefault) {
-                            $cloned = clone $propertyMetadata;
-                            $cloned->setter = null;
-                            $this->accessor->setValue($object, $cloned->defaultValue, $cloned, $this->context);
+                        if (true === $contextSpecificMetadata->hasDefault) {
```

Comment:
I'm afraid it can has some side effects if `$this->context->getPropertyNamingStrategy()` is null as we are not going to clone at all.

Reply:
Ah true. Should we clone always or leave this part code as it was before?

Reply:
I would left it as it used to be :)